File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        
        if root is None:
            
            return []
        
        que = deque([root])
        
        max_list = []
        
        while que:
            
            max_list.append(-sys.maxsize)
            
            for _ in range(len(que)):
                
                node = que.popleft()
                
                max_list[-1] = max(max_list[-1], node.val)
                
                node.left is not None and que.append(node.left)
                
                node.right is not None and que.append(node.right)
                
        return max_list
                    
                
                
                
                

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function largestValues(root: TreeNode | null): number[] {
    
    if(!root) return [];
    
    const maxList = [];
    
    const que = [root];
    
    while(que.length > 0){
        maxList.push(Number.MIN_SAFE_INTEGER);
        
        const length = que.length;
        
        for(let index = 0; index < length;++index){
            
            const node = que.shift();
            
            maxList[maxList.length-1] = Math.max(maxList.at(-1), node.val);
            node.left && que.push(node.left);
            node.right && que.push(node.right);
        }
        
    }
    
    return maxList;
    
    
    
    
};